scripts/oes-data-migration-scripts/migration_v3.12.x_to_v4.0.x.py

```diff
@@ -237,13 +237,13 @@ def update_custom_gates_navigation_url(plKeyExecDict):
                 print("The output is: ", output)
                 output_str = str(output.decode("utf-8"))
                 output_json = json.loads(output_str)
-                if b'approval' == stage_type:
+                if b'approval' == stage_type and 'navigationalURL' in output_json:
                     update_approval_gate_url(output_json, plKey, exec_str)
 
-                elif b'verification' == stage_type or b'testverification' == stage_type:
+                elif (b'verification' == stage_type or b'testverification' == stage_type) and ('verificationURL' in output_json or 'canaryReportURL' in output_json):
                     update_verification_gate_url(output_json, plKey, exec_str)
 
-                elif b'policy' == stage_type:
+                elif b'policy' == stage_type and 'policyName' in output_json:
```

Comment:
@Pranav-b-7 to me looks incorrect... inside update_policy_gate_url() we are trying to form the policyName when it is missing in the json_data.. please recheck